src/TrackedStorage.ts

```diff
@@ -0,0 +1,242 @@
+ * storage.setItem('count', 42);
+ * const count = storage.getItem('count'); // 42
+ * ```
+ */
+export class TrackedStorage {
+  #prefix: string;
+  #storage: Storage;
+  #cache: TrackedMap<string, unknown>;
+  #managedKeys: Set<string>;
+  #cacheKey: string;
+
+  constructor(storage: Storage, prefix?: string) {
+    this.#storage = storage;
+    this.#prefix = prefix ?? DEFAULT_PREFIX;
+    this.#cacheKey = getCacheKey(storage, this.#prefix);
+
+    // Get or create shared cache and managed keys for this storage+prefix combination
+    if (!sharedCaches.has(this.#cacheKey)) {
+      const cache = new TrackedMap<string, unknown>(new Map());
+      const managedKeys = new Set<string>();
+
+      sharedCaches.set(this.#cacheKey, cache);
+      sharedManagedKeys.set(this.#cacheKey, managedKeys);
+    }
+
+    this.#cache = sharedCaches.get(this.#cacheKey)!;
+    this.#managedKeys = sharedManagedKeys.get(this.#cacheKey)!;
+
+    // Sync cache with storage: remove cached keys that no longer exist in storage
+    for (const cachedKey of Array.from(this.#managedKeys)) {
+      if (storage.getItem(cachedKey) === null) {
+        this.#cache.delete(cachedKey);
+        this.#managedKeys.delete(cachedKey);
+      }
+    }
+
+    // Add any new keys from storage that match our prefix
+    for (let i = 0; i < storage.length; i++) {
+      const key = storage.key(i);
+      if (
+        key &&
+        key.startsWith(`${this.#prefix}:`) &&
+        !this.#managedKeys.has(key)
+      ) {
+        this.#cache.set(key, jsonParseAndFreeze(storage.getItem(key)));
+        this.#managedKeys.add(key);
+      }
+    }
+  }
+
+  /**
+   * Build the full storage key with prefix
+   */
+  #buildKey = (key: string): string => {
+    return `${this.#prefix}:${key}`;
+  };
+
+  /**
+   * Strip the prefix from a key
+   */
+  #stripPrefix = (key: string): string => {
+    const prefix = `${this.#prefix}:`;
+    return key.startsWith(prefix) ? key.slice(prefix.length) : key;
+  };
+
+  /**
+   * Get an item from storage. Returns the parsed value or null if not found.
+   * Objects and arrays are deep frozen to prevent mutation.
+   *
+   * Note: This method accesses the cache for reactivity. Values that don't exist
+   * in storage will return null without being cached to avoid tracking violations.
+   */
+  getItem = <T = unknown>(key: string): T | null => {
+    const prefixedKey = this.#buildKey(key);
+
+    // Check if the key is tracked in our cache
+    if (this.#cache.has(prefixedKey)) {
+      return this.#cache.get(prefixedKey) as T | null;
+    }
+
+    // For keys not in cache, check if they exist in storage
+    // If they don't exist, return null without caching to avoid tracking violations
+    const rawValue = this.#storage.getItem(prefixedKey);
+    if (rawValue === null) {
+      return null;
+    }
+
+    // Key exists in storage but not cache - this shouldn't normally happen
+    // since we initialize the cache in constructor, but handle it gracefully
```

Comment:
This comment makes me wonder if we actually need to initialize the cache with all values in storage before they are consumed? Maybe we can delay that to the time a value is actually read to improve performance?